pure.zsh

```diff
@@ -582,6 +583,29 @@ prompt_pure_setup() {
 	# to be available, it was added in Zsh 5.3.
 	autoload -Uz +X add-zle-hook-widget 2>/dev/null
 
+	# most used colors
+	typeset -gA prompt_pure_colors
+	prompt_pure_colors=(
+		branch		242
+		error		red
+		exec_time	yellow
+		git_arrow	cyan
+		host		242
+		path		blue
+		root		white
+		success		magenta
+		user		242
+		virtualenv  242
+	)
+	# modify default colors if set by the user
+	local color_temp
+	for key val in ${(kv)prompt_pure_colors}; do
+		if ! zstyle -T ":prompt:pure:$key" color; then
+			zstyle -s ":prompt:pure:$key" color color_temp
+			prompt_pure_colors[$key]=$color_temp
+		fi
+	done
```

Comment:
I really like the cleanness of this implementation, however, it doesn't look like it supports changing colors at runtime and I'm concerned with how easy it will be for our users to experiment.

For instance, if I type `zstyle :prompt:pure:success color green` in the terminal, nothing happens. I could type in `prompt pure` to re-init Pure, but it seems like an extra step we should be able to avoid.

Maybe we could re-run this at every preprompt before the colors are checked? Or would that create a lot of overhead?

Reply:
The response I got was one `zstyle` lookup is ~4µs so I'll rewrite the PR to only use styles.

Reply:
That sounds affordable 😄, thanks for investigating!